libraries/stumbler/src/main/java/org/mozilla/mozstumbler/service/stumblerthread/motiondetection/MotionSensor.java

```diff
@@ -138,10 +138,17 @@ public void stop() {
         Context mContext;
         private static final float DIST_THRESHOLD_M = 30.0f;
         private static final long TIME_INTERVAL_MS = 30000;
+        private Location mLastLocation;
 
         private LocationListener mNetworkLocationListener = new LocationListener() {
             public void onLocationChanged(Location location) {
-                AppGlobals.guiLogInfo("MotionSensor.NetworkLocationChangeDetector triggered.");
+                if (mLastLocation == null) {
+                    mLastLocation = location;
+                    return;
+                }
+
+                AppGlobals.guiLogInfo("MotionSensor.NetworkLocationChangeDetector triggered. (" +
+                    Math.round(location.distanceTo(mLastLocation)) + "m)");
                 Intent sendIntent = new Intent(ACTION_USER_MOTION_DETECTED);
```

Comment:
You need to set `mLastLocation` here as well or else you'll only set last location once.